test/prism_regression/range.rb

```diff
@@ -0,0 +1,12 @@
+# typed: true
+
+1..2 # Include end
+
+3...4 # Exclude end
+
+# Endless ranges need to be parenthesized
+("Has no end.."..)
+("Has no end..."...)
```

Comment:
I needed to detour an implement `PM_PARENTHESES_NODE`, in order to express this syntax here.